### PR TITLE
Add option for suppressing styler communication

### PR DESCRIPTION
**Pull request by michaelquinn32**
This implements FR #637

### CODE REVIEW

**Comment by michaelquinn32** [comment]
Hi Lorenz!

Thanks for the patience. I finally got around to this. One thing. I looked a little bit, but couldn't find if or where styler's communication was tested. Mind sharing a code pointer?

Thanks!

**Comment by lorenzwalthert** [comment]
Thanks. Two things to debate: 

- You implemented as a boolean option. This only allows two modes. Do we ever want to support more than two modes? I.e. moderately verbose?
- In {usethis}, the option is called `usethis.quiet`. I don't know if the tidyverse has embraced standardisation for options that manage communication. Maybe we should investigate that.

Tests for communication are here: https://github.com/r-lib/styler/blob/master/tests/testthat/test-public_api.R

**Comment by michaelquinn32** [comment]
I can't think of a situation where "less verbose" makes sense. For my use case, either the messaging is useful (which it almost always is), or it is interfering with something like stdout.

I'm happy to update the flag name. Alignment with other packages is good.

File: NEWS.md [hunk 1]
```diff
@@ -6,33 +6,35 @@
   when there are no comments before or after the blank line (#629, #630, #635).
 - `style_file()` and friends gain argument `dry` to control if changes should 
   be applied to files or not (#634).
+- added an option for disabling all communication when using the package
+  (styler.quiet) (#640).
```
Comment (lorenzwalthert) [inline NEWS.md:10]:
```suggestion
  (`styler.quiet`) (#640).
```

**Comment by lorenzwalthert** [comment]
How come that a significant part of `NEWS.md` got reformatted in 32f659c85cd8f812a9d616294cb46e33f2981de7?

**Comment by michaelquinn32** [comment]
Because I have an autoformatter that needs to be manually disabled :man_facepalming:.

Sorry, it's fixed.

**Comment by lorenzwalthert** [comment]
Out of curiosity: Which formatter do you use? Could we add that to {precommit} as a hook?

**Comment by lorenzwalthert** [comment]
Thanks @michaelquinn32.

**Comment by michaelquinn32** [comment]
A precommit would be great!

I'm using an internal formatter, but I don't think it matters all that much. prettier, for example, supports markdown: https://github.com/prettier/prettier

**Comment by lorenzwalthert** [comment]
They already made a pre-commit compatible hooks themself: https://prettier.io/docs/en/precommit.html#option-3-pre-commithttpsgithubcompre-commitpre-commit

**Pull request merged**